Softeq.XToolkit.Common.Tests/Extensions/DateTimeExtensionsTests/DateTimeExtensionsTests.cs

```diff
@@ -62,5 +80,110 @@ public void IsYesterday_ForDateTime_ReturnsExpectedValue(DateTime testDate, bool
+
+            var date = DateTime.Today;
+            var isInCurrentWeek = date.IsInCurrentWeek();
+            var count = isInCurrentWeek ? 1 : 0;
+
+            for (var i = 1; isInCurrentWeek; i++)
+            {
+                var previousDay = date.AddDays(-i);
+                var previousDayIsInCurrentWeek = previousDay.IsInCurrentWeek();
+                var nextDay = date.AddDays(i);
+                var nextDayIsInCurrentWeek = nextDay.IsInCurrentWeek();
+                count += previousDayIsInCurrentWeek ? 1 : 0;
+                count += nextDayIsInCurrentWeek ? 1 : 0;
+                isInCurrentWeek = previousDayIsInCurrentWeek || nextDayIsInCurrentWeek;
+            }
+
+            Assert.Equal(weekLength, count);
+        }
+
+        [Theory]
+        [MemberData(nameof(DateTimeExtensionsDataProvider.OutsideCurrentWeekData), MemberType = typeof(DateTimeExtensionsDataProvider))]
+        public void IsInCurrentWeek_ForOutsideCurrentWeekDays_ReturnsFalse(DateTime testDate)
+        {
+            var isInCurrentWeek = testDate.IsInCurrentWeek();
+            Assert.False(isInCurrentWeek);
+
+            var isInCurrentMondayWeek = testDate.IsInCurrentWeek(DayOfWeek.Monday);
+            Assert.False(isInCurrentMondayWeek);
+        }
+
+        [Fact]
+        public void IsInCurrentWeek_ForBoundaryWeekDaysWithDefaultParam_ReturnsCorrectValues()
+        {
+            var firstWeekDay = DateTime.Today.FirstDayOfWeek();
+            var firstDayIsInCurrentWeek = firstWeekDay.IsInCurrentWeek();
+            Assert.True(firstDayIsInCurrentWeek);
+
+            var previousToFirstWeekDay = firstWeekDay.AddDays(-1);
+            var previousToFirstDayIsInCurrentWeek = previousToFirstWeekDay.IsInCurrentWeek();
+            Assert.False(previousToFirstDayIsInCurrentWeek);
+
+            var lastWeekDay = DateTime.Today.LastDayOfWeek();
+            var lastDayIsInCurrentWeek = lastWeekDay.IsInCurrentWeek();
+            Assert.True(lastDayIsInCurrentWeek);
+
+            var nextToLastWeekDay = lastWeekDay.AddDays(1);
+            var nextToLastDayIsInCurrentWeek = nextToLastWeekDay.IsInCurrentWeek();
+            Assert.False(nextToLastDayIsInCurrentWeek);
+        }
+
+        [Theory]
+        [InlineData(DayOfWeek.Monday, DayOfWeek.Sunday)]
+        [InlineData(DayOfWeek.Wednesday, DayOfWeek.Tuesday)]
+        [InlineData(DayOfWeek.Saturday, DayOfWeek.Friday)]
+        public void IsInCurrentWeek_ForBoundaryWeekDaysWithSpecifiedParam_ReturnsCorrectValues(DayOfWeek startDayOfWeek, DayOfWeek endDayOfWeek)
+        {
+            var firstWeekDay = DateTime.Today.FirstDayOfWeek(startDayOfWeek);
+            var firstDayIsInCurrentWeek = firstWeekDay.IsInCurrentWeek(startDayOfWeek);
+            Assert.True(firstDayIsInCurrentWeek);
+
+            var previousToFirstWeekDay = firstWeekDay.AddDays(-1);
+            var previousToFirstDayIsInCurrentWeek = previousToFirstWeekDay.IsInCurrentWeek(startDayOfWeek);
```

Comment:
looks like can be split to separate tests with concrete DateTime

Reply:
Reworked